src/components/highlight/highlight.stories.tsx

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { Meta, StoryObj } from '@storybook/react';
+
+import { EuiHighlight, EuiHighlightProps } from './highlight';
+
+const meta: Meta<EuiHighlightProps> = {
+  title: 'EuiHighlight',
+  component: EuiHighlight,
+  argTypes: {},
+  // Component defaults
+  args: {
+    search: 'quick',
+    strict: false,
+    highlightAll: false,
+    hasScreenReaderHelpText: true,
+  },
+};
+
+export default meta;
+type Story = StoryObj<EuiHighlightProps>;
+
+export const Playground: Story = {
+  args: {
+    children: 'The quick brown fox jumped over the lazy dog.',
+  },
+};
```

Comment:
This is one of those components with a special enough prop use-case I think we could add an extra story here to help us QA it! The use case is when `search` is an array of strings, e.g.:

```suggestion
};

export const SearchArray: Story = {
  children: 'The quick brown fox jumped over the lazy dog.',
  search: ['fox', 'dog'],
  highlightAll: true,
};
```